profiling/src/profile/internal/mapping.rs

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::{Id, Item, PprofItem};
+use crate::profile::{pprof, StringId};
+use std::num::NonZeroU32;
+
+#[derive(Eq, PartialEq, Hash)]
+pub struct Mapping {
+    /// Address at which the binary (or DLL) is loaded into memory.
+    pub memory_start: u64,
```

Comment:
The PR description suggested something about squeezing this into a u32, but I think it needs to be u64.  Realistically you can bound userspace addresses to some range on the order of 2^47 or whatever, but I _think_ we'd really like to add kernel profiling for the native profiler, which would boost us back up to u64.